src/executor/update_executor.cpp

```diff
@@ -290,6 +295,14 @@ bool UpdateExecutor::DExecute() {
             }
             transaction_manager.SetTransactionResult(current_txn,
                                                      ResultType::FAILURE);
+            if (fk_failure) {
+              throw ConstraintException("FOREIGN KEY constraint violated : " +
+                               std::string(new_tuple.GetInfo()));
+            } else {
+              throw ConstraintException("Secondary index constraint violated : " +
```

Comment:
Yes, these ConstraintExceptions are caught in the traffic cop, which then updates the transaction result appropriately.